### PR TITLE
feat: set up Coinbase 2FA UI

### PR DESCRIPTION
**Pull request by ClaudeHangui**
Prior to completing a transaction on Coinbase, a user is required to confirm his identity using a 2FA code. This story focuses on providing the design of the screen where the user has to enter his auth code
## Issue being fixed or feature implemented
- replace the initial UI used to display the 2FA code 
- update error handling flow for BuyDash/BuySwap flows
- renaming of some properties and/or functions to keep a certain consistency

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [x] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: common/src/main/res/drawable/ic_red_circle.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
...
+</vector>
```
Comment (Syn-McJ) [inline common/src/main/res/drawable/ic_red_circle.xml:5]: This isn't just a circle but an exclamation mark, so this file might use a rename.

File: common/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -863,4 +863,24 @@
         <item name="strokeColor">@android:color/transparent</item>
         <item name="strokeWidth">0dp</item>
     </style>
+    <style name="Headline6.SemiBold">
+        <item name="fontFamily">@font/montserrat_semibold</item>
+    </style>
```
Comment (Syn-McJ) [inline common/src/main/res/values/styles.xml:868]: Headline style should go into the typography section

File: common/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -863,4 +863,24 @@
...
+        <item name="backgroundColor">@color/white</item>
+        <item name="strokeColor">@color/dash_blue</item>
+        <item name="cornerRadius">10dp</item>
+    </style>
+    <style name="Caption.DarkSteelGray700">
+        <item name="android:textColor">@color/dark_steel_gray_700</item>
+    </style>
+    <style name="Caption.Red300">
+        <item name="android:textColor">@color/red_300</item>
+    </style>
```
Comment (Syn-McJ) [inline common/src/main/res/values/styles.xml:879]: Same as for the headline

File: integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/ui/CoinbaseBuyDashOrderReviewFragment.kt [hunk 4]
```diff
@@ -63,7 +58,6 @@ class CoinbaseBuyDashOrderReviewFragment : Fragment(R.layout.fragment_coinbase_b
     private lateinit var selectedPaymentMethodId: String
     private var loadingDialog: FancyAlertDialog? = null
     private var isRetrying = false
-    private var transactionStateDialog: CoinBaseBuyDashDialog? = null
     private var newBuyOrderId: String? = null
     @Inject
     lateinit var analyticsService: FirebaseAnalyticsServiceImpl
```
Comment (Syn-McJ) [inline integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/ui/CoinbaseBuyDashOrderReviewFragment.kt:63]: Not the current change, but we should be injecting the interface and not the implementation.

File: integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/viewmodels/CoinbaseBuyDashOrderReviewViewModel.kt [hunk 5]
```diff
@@ -43,28 +44,24 @@ class CoinbaseBuyDashOrderReviewViewModel @Inject constructor(
...
     var sendFundToWalletParams: SendTransactionToWalletParams ? = null
     val placeBuyOrderFailedCallback = SingleLiveEvent<String>()
     private val _placeBuyOrder: MutableLiveData<PlaceBuyOrderUIModel> = MutableLiveData()
     val placeBuyOrder: LiveData<PlaceBuyOrderUIModel>
         get() = _placeBuyOrder
 
-    private val _commitBuyOrderSuccessCallback: MutableLiveData<SendTransactionToWalletParams> = MutableLiveData()
-    val commitBuyOrderSuccessCallback: LiveData<SendTransactionToWalletParams>
+    private val _commitBuyOrderSuccessCallback: MutableLiveData<Event<SendTransactionToWalletParams>> = MutableLiveData()
+    val commitBuyOrderSuccessCallback: LiveData<Event<SendTransactionToWalletParams>>
```
Comment (Syn-McJ) [inline integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/viewmodels/CoinbaseBuyDashOrderReviewViewModel.kt:56]: Isn't the `SingleLiveEvent` serve the same purpose as `LiveData<Event<>>`? Not sure we need 2 different solutions

**Review by Syn-McJ — APPROVED** [review]
Looks good, just some minor comments.

**Review by Syn-McJ — APPROVED** [review]

**Review by HashEngineering — APPROVED** [review]

**Pull request merged**